### PR TITLE
Propagate more errors to the client side

### LINKED ISSUE #444: Improve client side error messages
### Feature description

Currently client side errors are something like ${auto.test.resources.datasources.default.url} cannot be resolved.

We should improve these errors somehow to indicate why:

* Docker is not running?
* Test resource server not running?
* Docker is running but container startup failed -> show error

### PR DESCRIPTION
**Pull request by melix**
This commit improves error reporting on the client side. It propagates some errors which were previously only captured on the server side. In particular, it will now:

- report which property was being resolved, instead of showing internal `auto.test.resources` properties
- capture errors which are sent by the server but were dropped because of lack of JSON error handling
- report when a container cannot be started or failed during startup
- provide clearer message when the test resources service is down

Fixes #444

### CODE REVIEW

File: test-resources-client/src/main/java/io/micronaut/testresources/client/DefaultTestResourcesClient.java [hunk 1]
```diff
@@ -51,6 +53,8 @@ public class DefaultTestResourcesClient implements TestResourcesClient {
     private static final Argument<List<String>> LIST_OF_STRING = Argument.LIST_OF_STRING;
     private static final Argument<String> STRING = Argument.STRING;
     private static final Argument<Boolean> BOOLEAN = Argument.BOOLEAN;
+    private static final String INTERNAL_SERVER_ERROR_PREFIX = "Internal Server Error: ";
+    private static final String INTERNAL_SERVER_ERROR = "Internal Server Error";
```
Comment (alvarosanchez) [inline test-resources-client/src/main/java/io/micronaut/testresources/client/DefaultTestResourcesClient.java:57]:
```suggestion
    private static final String INTERNAL_SERVER_ERROR = "Internal Server Error";
    private static final String INTERNAL_SERVER_ERROR_PREFIX = INTERNAL_SERVER_ERROR + ": ";    
```

File: test-resources-client/src/main/java/io/micronaut/testresources/client/SimpleJsonErrorModel.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+package io.micronaut.testresources.client;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.micronaut.serde.annotation.Serdeable;
+
+import java.util.List;
+
+@Serdeable
+public record SimpleJsonErrorModel(
+    @JsonProperty("message") String message,
```
Comment (alvarosanchez) [inline test-resources-client/src/main/java/io/micronaut/testresources/client/SimpleJsonErrorModel.java:25]: Is the annotation needed here?

File: test-resources-client/src/main/java/io/micronaut/testresources/client/TestResourcesClientPropertyExpressionResolver.java [hunk 3]
```diff
@@ -108,7 +109,26 @@ public <T> Optional<T> resolve(PropertyResolver propertyResolver,
         }
 
         private static Optional<String> callClient(String expression, TestResourcesClient client, Map<String, Object> props, Map<String, Object> properties) {
-            return client.resolve(expression, props, properties);
+            return withErrorHandling(
+                () -> client.resolve(expression, props, properties),
+                () -> "Test resources service wasn't able to revolve expression '" + expression + "'"
```
Comment (alvarosanchez) [inline test-resources-client/src/main/java/io/micronaut/testresources/client/TestResourcesClientPropertyExpressionResolver.java:114]:
```suggestion
                () -> "Test resources service wasn't able to revolve expression '%s'".formatted(expression)
```

**Review by alvarosanchez — APPROVED** [review]

File: test-resources-client/src/main/java/io/micronaut/testresources/client/SimpleJsonErrorModel.java (hunk 2 above)
Comment (melix) [reply test-resources-client/src/main/java/io/micronaut/testresources/client/SimpleJsonErrorModel.java:25]: It shouldn't, kept it for consistency with the other properties

**Pull request merged**